charts/common/templates/service-cloudarmor.yaml

```diff
@@ -7,9 +7,10 @@ metadata:
     {{- include "common.labels" . | nindent 4 }}
   annotations:
     {{- if .Values.cloudArmor.backendConfig.iap }}
-    cloud.google.com/backend-config: '{"ports": {"80":"{{ include "common.name" . }}", "443":"{{ include "common.name" . }}"},"default": "{{ include "common.name" . }}"}'
+    cloud.google.com/backend-config: '{"ports": {"80":"{{ include "common.name" . }}", "443":"{{ include "common.name" . }}"},"default": "{{ include "common.name" . }}", "customRequestHeaders": {{ .Values.cloudArmor.backendConfig.customHeaders | toJson }}}'
```

Comment:
@sjgupta19 I'm not sure toJson is a valid type conversion, Did you mean `toYaml` instead?

Reply:
seems like it is valid function -  https://helm.sh/docs/chart_template_guide/function_list/

Reply:
@sjgupta19 ic, Can you share the helm template --dry-run where you tested this? Just to make sure does not error out

Reply:
hmm.. erroring out -- `Error: file '/Users/srajangupta/Desktop/dave-github/charts/charts/common/templates/service-cloudarmor.yaml' seems to be a YAML file, but expected a gzipped archive`

Reply:
thats the same error I am getting without these changes. So I guess its unrelated to my changes?

Reply:
you need to test on a particular SRE chart that can include those new values that you defined on the template/values

Reply:
ohh I don't see where are you changing the version of the chart to refer your changes:
https://github.com/dave-inc/charts/blob/master/charts/common/Chart.yaml#L18

Reply:
I am curious if there is any other way we can add these custom headers? I am following this guide - https://cloud.google.com/load-balancing/docs/https/custom-headers-global

Reply:
just bumped the chart version

Reply:
here is the documentation on the annotation you are modifying, doesn't look like it supports that directive https://cloud.google.com/kubernetes-engine/docs/how-to/ingress-configuration

Reply:
looks like you can use the backendconfig for this https://cloud.google.com/kubernetes-engine/docs/how-to/ingress-configuration#request_headers

Reply:
So do i have to specifically mention that in backendConfig? isn't `.Values.cloudArmor.backendConfig.customHeaders` this essentially adding it to backendConfig?